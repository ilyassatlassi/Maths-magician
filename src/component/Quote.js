import React, { useEffect, useState } from "react";

function Quote() {
  const [initQuote, setQuote] = useState({});

  const API_KEY = "1d3V+eGyiKNzKIrm5ZX9Ig==uBvxDSWkiDzhs17K";

  useEffect(() => {
    const fetchData = async () => {
      try {
        const res = await fetch(
          "https://api.api-ninjas.com/v1/quotes?category=money",
          {
            headers: { "X-Api-Key": API_KEY },
          });
          const json = await res.json();
          setQuote(json[0]);
      } catch (error){
        setQuote({ quote: `${error}` });
      }
    };
    fetchData();
  }, []);
  return !initQuote.quote ? (
    <div className="Quote">
      <div className="ring"></div>
      <span>loading...</span>
    </div>
  ) : initQuote.author ? (
    <div className="Quote-section">
      <p className="author">&quot;{initQuote.author}&quot;</p>
      <p className="Quote-phrase">{initQuote.quote}</p>
    </div>
  ) : (
    <div>
      <p className="Error">{initQuote.quote}</p>
    </div>
  );
}

export default Quote;
