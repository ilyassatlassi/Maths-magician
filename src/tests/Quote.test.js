import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quote from '../component/pages/Quote';
import '@testing-library/jest-dom/extend-expect';

test('Should render Quotes Component', () => {
    render(<Quote />);
    const HomeElement = screen.getByTestId('Quote');
    expect(HomeElement).toBeInTheDocument();
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});