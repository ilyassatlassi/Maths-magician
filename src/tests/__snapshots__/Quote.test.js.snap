// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Quotes Component 1`] = `
<div
  data-testid="Quote"
>
  <div
    className="Quote"
  >
    <div
      className="ring"
    />
    <span>
      loading...
    </span>
  </div>
</div>
`;
